### ISSUE TITLE
False positives on "Declare only one React component per file" (react/no-multi-comp)

### ISSUE THREAD

**Issue by MoOx** [yannickcr/eslint-plugin-react#1088]
```js
import test from "jest-ava-api"
import React, { createElement } from "react"
import { createRenderer } from "react-addons-test-utils"

const renderer = (...args) => { // I guess this is detected as a component
  const render = createRenderer()
  render.render(...args)
  return render.getRenderOutput()
}

test("what", () => {
  return {
    isActive: () => false, // what?
  }
})
```

react/no-multi-comp is detecting 2 components here since I get a warning on the last () => false.
Both are wrong.

Ps: the code does not make sense as it is as I removed some part to make the debugging more easy.

Not that if I does not import createElement, the error seems to disappear  (but I need this...)

**Comment by ljharb** [comment]
(Unrelated, but you may want to check out enzyme rather than creating your own renderers)

**Comment by nickbouton** [comment]
@benstepp @ljharb I seem to be having this problem again with v7.0.0. Any component method that has a name starting with `render...` and returns something non-null appears to falsely trigger this rule, e.g.:

![image](https://cloud.githubusercontent.com/assets/169988/25790395/014eb01c-336d-11e7-91cf-8ca3dd3a8a68.png)

**Comment by ljharb** [comment]
@nickbouton would you mind filing a new issue about it?

**Comment by nickbouton** [comment]
@ljharb Sure, added #1181 & removed my comment here.
